clang/test/dpct/cub/intrinsic/shl_add.cu

```diff
@@ -0,0 +1,40 @@
+// UNSUPPORTED: cuda-8.0, cuda-9.0, cuda-9.1, cuda-9.2, cuda-10.0, cuda-10.1, cuda-10.2
+// UNSUPPORTED: v8.0, v9.0, v9.1, v9.2, v10.0, v10.1, v10.2
+// RUN: dpct -in-root %S -out-root %T/intrinsic/shl_add %S/shl_add.cu --cuda-include-path="%cuda-path/include" -- -std=c++14 -x cuda --cuda-host-only
+// RUN: FileCheck --input-file %T/intrinsic/shl_add/shl_add.dp.cpp --match-full-lines %s
+
+// CHECK:#include <sycl/sycl.hpp>
+// CHECK:#include <dpct/dpct.hpp>
+#include <cub/cub.cuh>
+#include <stdio.h>
+
+// CHECK:void kernel1(int *res) {
+// CHECK-NEXT:  *res = (((unsigned int)1 << (unsigned int)2) + (unsigned int)3);
+// CHECK-NEXT:}
+__global__ void kernel1(int *res) {
+  *res = cub::SHL_ADD(1, 2, 3);
+}
+
+// CHECK:void kernel2(int *res, unsigned a, int b, unsigned c) {
+// CHECK-NEXT:  *res = ((a << (unsigned int)b) + c);
+// CHECK-NEXT:}
+__global__ void kernel2(int *res, unsigned a, int b, unsigned c) {
```

Comment:
there is clamp logic there of SHL_ADD,   need extra option on parameter B. 
asm ("vshl.u32.u32.u32.clamp.add %0, %1, %2, %3;" :
        "=r"(ret) : "r"(x), "r"(shift), "r"(addend));

For test purpose, suggest to add help function for SHL_ADD.

Reply:
SHL/SHR have moved to a seperate PR.